subprojects/packagefiles/glfw/meson.build

```diff
@@ -0,0 +1,259 @@
+if (sys_os == 'linux' or sys_os == 'sunos')
+  foreach l : ['m', 'rt', 'dl']
+    deps += sys_cc.find_library(l, required : true)
+  endforeach
+  deps += dependency('threads')
+elif sys_os == bsd
+  foreach l : ['m', 'rt']
+    deps += sys_cc.find_library(l, required : true)
+  endforeach
+  deps += dependency('threads')
+endif
+
+
+### Configuration ###
+cfg_data = configuration_data()
+foreach d : ['osmesa', 'x11', 'wayland', 'win32', 'cocoa']
+  cfg_data.set('_GLFW_' + d.to_upper(), opt_display == d)
+endforeach
+cfg_file = configure_file(configuration : cfg_data
+                         ,output : 'glfw_config.h')
+
+
+### Targets ###
+#@ Install key
+I = opt_install
+#@ Primary target:
+##   GLFW library
+#  Common files
+s_common = [cfg_file
+           ,'src/context.c'
+           ,'src/init.c'
+           ,'src/input.c'
+           ,'src/monitor.c'
+           ,'src/vulkan.c'
+           ,'src/window.c']
+#  Backend files
+if opt_display == 'osmesa'
+  s_display = ['src/null_init.c'
+              ,'src/null_monitor.c'
+              ,'src/null_window.c'
+              ,'src/null_joystick.c'
+              ,'src/posix_time.c'
+              ,'src/posix_thread.c'
+              ,'src/osmesa_context.c']
+elif opt_display == 'x11'
+  s_display = ['src/x11_init.c'
+              ,'src/x11_monitor.c'
+              ,'src/x11_window.c'
+              ,'src/xkb_unicode.c'
+              ,'src/posix_time.c'
+              ,'src/posix_thread.c'
+              ,'src/osmesa_context.c'
+              ,'src/egl_context.c'
+              ,'src/glx_context.c']
+  if sys_os == 'linux'
+    s_display += 'src/linux_joystick.c'
+  else
+    s_display += 'src/null_joystick.c'
+  endif
+elif opt_display == 'wayland'
+  s_display = [s_wayland
+              ,'src/wl_init.c'
+              ,'src/wl_monitor.c'
+              ,'src/wl_window.c'
+              ,'src/xkb_unicode.c'
+              ,'src/posix_time.c'
+              ,'src/posix_thread.c'
+              ,'src/osmesa_context.c'
+              ,'src/egl_context.c']
+  if sys_os == 'linux'
+    s_display += 'src/linux_joystick.c'
+  else
+    s_display += 'src/null_joystick.c'
+  endif
+elif opt_display == 'win32'
+#——TODO:
+# ☐ Implement Win32 build routine
+  s_display = ['src/win32_init.c'
+              ,'src/win32_joystick.c'
+              ,'src/win32_monitor.c'
+              ,'src/win32_time.c'
+              ,'src/win32_thread.c'
+              ,'src/win32_window.c'
+              ,'src/wgl_context.c']
+elif opt_display == 'cocoa'
+#——TODO:
+# ☐ Implement Cocoa build routine
+  s_display = ['src/cocoa_init.m'
+              ,'src/cocoa_joystick.m'
+              ,'src/cocoa_monitor.m'
+              ,'src/cocoa_window.m'
+              ,'src/cocoa_time.c'
+              ,'src/posix_thread.c'
+              ,'src/nsgl_context.m']
+endif
+srcfiles = [s_common, s_display]
+incdirs = [include_directories('include'), include_directories('src')]
+#  Build
+glfw_lib = build_target('glfw3'
```

Comment:
Why use build_target instead of directly using the type? The purpose of this function is when you need to dynamically choose which target_type to use.